internal/cmd/commands/credentialstorescmd/vault_funcs.go

```diff
@@ -0,0 +1,190 @@
+	flagTlsSkipVerify bool
+}
+
+func extraVaultActionsFlagsMapFuncImpl() map[string][]string {
+	flags := map[string][]string{
+		"create": {
+			addressFlagName,
+			namespaceFlagName,
+			vaultCaCertFlagName,
+			tlsServerNameFlagName,
+			tlsSkipVerifyFlagName,
+			vaultTokenFlagName,
+			clientCertificateFlagName,
+			clientCertificateKeyFlagName,
+		},
+	}
+	flags["update"] = flags["create"]
+	return flags
+}
+
+func extraVaultFlagsFuncImpl(c *VaultCommand, set *base.FlagSets, _ *base.FlagSet) {
+	f := set.NewFlagSet("Vault Credential Store Options")
+
+	for _, name := range flagsVaultMap[c.Func] {
+		switch name {
+		case addressFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   addressFlagName,
+				Target: &c.flagAddress,
+				Usage:  "The address of the Vault server. This should be a complete URL such as https://127.0.0.1:8200",
+			})
+		case namespaceFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   namespaceFlagName,
+				Target: &c.flagNamespace,
+				Usage:  "The vault namespace the store should use.",
+			})
+		case vaultCaCertFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   vaultCaCertFlagName,
+				Target: &c.flagCaCert,
+				Usage:  "The CA Cert to use when connecting to vault. This can be the value itself, refer to a file on disk (file://) from which the value will be read, or an env var (env://) from which the value will be read.",
+			})
+		case tlsServerNameFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   tlsServerNameFlagName,
+				Target: &c.flagTlsServerName,
+				Usage:  `Name to use as the SNI host when connecting via TLS.`,
+			})
+		case tlsSkipVerifyFlagName:
+			f.BoolVar(&base.BoolVar{
+				Name:   tlsSkipVerifyFlagName,
+				Target: &c.flagTlsSkipVerify,
+				Usage:  "Whether to skip tls verification.",
+			})
+		case vaultTokenFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   vaultTokenFlagName,
+				Target: &c.flagVaultToken,
+				Usage:  "The vault token to use when boundary connects to vault for this store.",
+			})
+		case clientCertificateFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   clientCertificateFlagName,
+				Target: &c.flagClientCert,
+				Usage:  "The client certificate to use when boundary connects to vault for this store. This can be the value itself, refer to a file on disk (file://) from which the value will be read, or an env var (env://) from which the value will be read.",
+			})
+		case clientCertificateKeyFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   clientCertificateKeyFlagName,
+				Target: &c.flagClientCertKey,
+				Usage:  `The client certificate's private key to use when boundary connects to vault for this store. This can be the value itself, refer to a file on disk (file://) from which the value will be read, or an env var (env://) from which the value will be read.`,
+			})
+		}
+	}
+}
+
+func extraVaultFlagHandlingFuncImpl(c *VaultCommand, f *base.FlagSets, opts *[]credentialstores.Option) bool {
+	switch c.flagAddress {
```

Comment:
Can address and token not be provided? I just ask because for any values that can't be blank, we should skip generating a Default function for it. @talanknight